app/controllers/concerns/vendor_api/application_data_concerns.rb

```diff
@@ -0,0 +1,33 @@
+module VendorAPI
+  module ApplicationDataConcerns
```

Comment:
:+1: Nice to have this common logic shareable across controllers, I had similar issues with notes and `GetApplicationChoicesForProviders` but this becomes much neater when wrapped in a concern.